src/EditorFeatures/CSharp/SplitComment/SplitCommentCommandHandler.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.Editor.Implementation.SplitComment;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Microsoft.VisualStudio.Text.Operations;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.SplitComment
+{
+    [Export(typeof(ICommandHandler))]
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    [Name(nameof(SplitCommentCommandHandler))]
+    [Order(After = PredefinedCompletionNames.CompletionCommandHandler)]
+    internal partial class SplitCommentCommandHandler : AbstractSplitCommentCommandHandler
+    {
+        [ImportingConstructor]
+        public SplitCommentCommandHandler(
+            ITextUndoHistoryRegistry undoHistoryRegistry,
+            IEditorOperationsFactoryService editorOperationsFactoryService)
+        {
+            _undoHistoryRegistry = undoHistoryRegistry;
+            _editorOperationsFactoryService = editorOperationsFactoryService;
+        }
+
+        protected override bool LineContainsComment(ITextSnapshotLine line, int caretPosition)
+        {
+            var snapshot = line.Snapshot;
+            var text = snapshot.GetText();
+            _hasSpaceAfterComment = text.Contains(CommentSplitter.CommentCharacter + ' ');
```

Comment:
this is super expensive afaict.  you're getting the *entire* text of the file, and looking for `// ` in it.

Reply:
also... are you mutating instance state?

Reply:
Doesn't this just get the snapshot of the line that the caret is currently positioned in?